src/objects/Earth/index.js

```diff
@@ -43,18 +50,78 @@ export class Earth {
 
       const points = collectPoints(data);
 
+      const locationPointGroups = [];
       for (let i = 0; i < points[0].length; i += 1) {
         const locationPointGroup = new THREE.Group();
         locationPointGroup.name = `Location__${i}`;
         locationPointGroup.add(points[0][i]); // point
         locationPointGroup.add(points[1][i]); // ring
         locationPointGroup.add(points[2][i]); // hitbox
         scene.add(locationPointGroup);
+        locationPointGroups.push(locationPointGroup);
       }
 
       for (let i = 0; i < TOTAL_ARCS; i += 1) {
         drawArc(drawCurve(...getRandomPointPositions()));
       }
+
+      const camera = getCamera();
+      let current;
+      const delay = MARKER_AUTO_SELECT_DELAY / 100; // Normalize Millisecond Config to "tick count"
+      let lastContent;
+      let lastUpdated = 0;
+      const raycaster = new THREE.Raycaster();
+      setInterval(() => {
+        // eslint-disable-next-line prefer-destructuring
+        current = window.document.getElementsByClassName('location visible')[0];
+        if (current && current.childNodes) {
+          const content = current.textContent;
+          if (content !== lastContent) {
+            lastContent = content;
+            lastUpdated = 0;
+            return;
+          }
+        }
+
+        lastUpdated += 1;
+
+        // Wait x Seconds since Last Location Popup Change
+        if (lastUpdated > delay) {
+          onAutoUpdate();
+        }
+      }, 100);
+
+      function onAutoUpdate() {
+        // Pick a Random Location
+        const random = Math.round(Math.random() * (locationPointGroups.length - 1));
+        const group = locationPointGroups[random];
+        const selection = window.document.getElementById(`Location__${random}`);
+        const a = camera.position.normalize();
+        const b = group.children[2].position.normalize();
```

Comment:
I think it would have been more ideal to position the Group in an absolute manner, and then we would apply relative positions to the children of the Group.

But as is, the Group has a position of `0,0,0`, and all objects in the Group are positioned "absolutely" (relative from 0,0,0). This threw me through a HUGE loop when I was trying to use the Group's position as the target of the raycast, which was always going to `0,0,0` 😆 Just food for thought, maybe I'll refactor it later. 🐒